py-polars/src/lazyframe.rs

```diff
@@ -165,16 +165,12 @@ impl PyLazyFrame {
         encoding: Wrap<CsvEncoding>,
         row_count: Option<(String, IdxSize)>,
         try_parse_dates: bool,
-        eol_char: &str,
+        eol_char: u8,
```

Comment:
I prefer doing the parsing in the Rust binding, rather than calling `ord` in Python. Any processing we can do in Rust should be done in Rust.

Reply:
well, I only aligned these, so I would then reverse the `eol_char: u8` which are there to `eol_char: &str` instead?

I personally think it's cleaner for the API to not need to do any parsing at all, and just expect the correct type directly. Hence my choice to request a byte instead of a string.

WDYT?

Reply:
> well, I only aligned these, so I would then reverse the eol_char: u8 which are there to eol_char: &str instead?

Yes, I know it's done like this elsewhere, but that should indeed be changed.

As I said, any parsing we can do in Rust should be done in Rust. The API of `PySeries` should be as close to `Series` as possible. In this case, both should accept a string (instead of one accepting a string and the other accepting an integer).

Reply:
Just to confirm:
py-polars should accept a str, whilst polars should accept u8?

Reply:
For now.

I'd say the Rust side should probably work with chars here. Passing an integer as a quote char is not very ergonomic. But let's keep that out of this PR for now.

Reply:
OK, let's re-iterate then as I'm visiting a lot of stuff within CSV here:

shall I make the sink/write/scan/read all accept strings instead of bytes?

I think it's best if I do this in one go as I'm now quite familiar with the underlying code. And I think it's easier if I do this now than that someone else would do it in another go?
WDYT?

Reply:
Another thought occurred to me as well: copying a byte is much less processing intensive than copying a String (a pointer copy would be the same than an int).

I don't know how rust handles its memory (yet), so maybe you can take this into account because I'm fiddling with the u8 -> &str/String right now and it's not too difficult. Just a grind. But I want to make sure I'm on the right track here.

Reply:
And yet another thought (I'm on a roll!!): shouldn't we just make the python interface `quote_char = None`? Because then python needs to transport only None to rust, and not a string. As the processing would happen on the rust side anyhow to take its default.

However, this will make the python API (much) less expressive.

Just thought to throw the idea in here.

Reply:
Actually, shouldn't all these parameter accept `bytes` instead of `str`?

Then the PyO3 function can accept `&[u8]` (or `Vec<u8>`?), and assert length == 1 and extract the first element.

And then pass `u8 as char` to the Rust side.

I think that would be best, but not sure about that last part, maybe there are some benefits to using u8 directly.

> I think it's best if I do this in one go

This PR is already quite big. Let's leave these refactors for a separate PR, as now we're getting into territory that might be breaking / affect implementation details, rather than just simple renaming.

Reply:
Ok, reverted the types as you requested (it is indeed getting quite big!!).
Splitted off this discussion into a new ticket: #11600